File: backend/src/main/java/Money/Contract/ContractDTO.java
package Money.Contract;

import Money.Contract.Enum.ContractStatus;
import Money.Contract.Enum.ContractType;
import Money.Contract.Enum.Contractor;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ContractDTO {

    private Long id;

    private String name; // Наименование договора

    private String contractNumber; // Номер договора

    private LocalDate contractDate; // Дата договора

    private ContractStatus status; // Статус договора

    private ContractType type; // Тип договора

    private Contractor contractor; // Подрядчик

    private LocalDate startDate; // Дата начала договора

    private LocalDate endDate; // Дата окончания договора

    private BigDecimal plannedCostWithoutVAT; // Плановая стоимость без НДС

    private BigDecimal plannedVAT; // Плановая НДС

    private BigDecimal actualCostWithoutVAT; // Фактическая стоимость без НДС

    private BigDecimal actualVAT; // Фактический НДС

    private BigDecimal warrantyReserve; // Гарантийный резерв (%)

    private BigDecimal plannedAdvancePercent; // Плановый аванс (%)

    private BigDecimal plannedAdvance; // Плановый аванс (руб), рассчитывается на бэкенде

    private BigDecimal plannedCost; // Плановая стоимость, рассчитывается на бэкенде

    private BigDecimal actualCost; // Фактическая стоимость, рассчитывается на бэкенде

    private BigDecimal actualAdvance; // Фактический аванс (руб)
}

File: frontend/src/contract/DTO/ContractDTO.ts
// ContractDTO.ts

export enum ContractStatus { //вынести в справочник,
    DRAFT = 'Черновик',
    ACTIVE = 'Действует',
    COMPLETED = 'Выполнен',
    CANCELLED = 'Отменён',
}

export enum ContractType { //вынести в справочник,
    PIR = 'ПИР',
    SMR = 'СМР',
    OTHER = 'Прочие'
}

export enum Contractor { //вынести в справочник, заявать на типы контрактов
    CONTRACTOR_A = 'Подрядчик по СМР',
    CONTRACTOR_B = 'Подрядчик по ПИР',
    CONTRACTOR_C = 'Подрядчик по прочим'
}

export interface ContractDTO {
    id: number;
    name: string;                      // Наименование договора
    contractNumber: string;             // Номер договора
    contractDate: string;               // Дата договора
    status: ContractStatus;             // Статус договора
    type: ContractType;                 // Тип договора
    contractor: Contractor;             // Подрядчик
    startDate?: string;                 // Дата начала договора
    endDate?: string;                   // Дата окончания договора
    plannedCostWithoutVAT: number;      // Плановая стоимость без НДС
    plannedVAT: number;                 // Плановая НДС
    actualCostWithoutVAT?: number;       // Фактическая стоимость без НДС
    actualVAT?: number;                  // Фактический НДС
    warrantyReserve?: number;           // Гарантийный резерв (%)
    plannedAdvancePercent?: number;     // Плановый аванс (%)
    plannedAdvance?: number;            // Плановый аванс (рассчитывается на бэкенде)
    plannedCost?: number;               // Плановая стоимость (рассчитывается на бэкенде)
    actualCost?: number;                // Фактическая стоимость (рассчитывается на бэкенде)
    actualAdvance?: number;             // Фактический аванс
}
